homework/console/02/control.ts

```diff
@@ -0,0 +1,126 @@
+interface DoingInterface {
+  category: string;
+  content: string[];
+}
+interface MemoryInterface {
+  name: string;
+  gender: string;
+  age: number;
+  isStudent: boolean;
+  doing?: DoingInterface[];
+  hobby?: string[];
+}
+type ConvertedGender = { [key in Gender]: string };
+
+type paramsType = MemoryInterface | string | number;
+
+type Gender = 'female' | 'male';
+
+type GameType = 'game' | 'study' | 'memory';
+
+type GameStatusType = 'start' | 'pause' | 'stop';
+
+const convertGender: ConvertedGender = {
+  female: '여성',
+  male: '남성',
+};
+
+const GAME_MESSAGE = {
+  START: '게임이 시작되었습니다!',
+  PAUSE: '게임이 중지되었습니다!',
+  STOP: '게임이 종료되었습니다!',
+};
+
+let arr: number[] = [];
+
+const getGameStatus: { [key: string]: string } = {
+  start: GAME_MESSAGE.START,
+  pause: GAME_MESSAGE.PAUSE,
+  stop: GAME_MESSAGE.STOP,
+};
+
+const control = (type: GameType, params: paramsType) => {
+  let gameStatus = '';
+  let result: string | number[] = '';
+
+  if (type === 'game') {
+    console.log(params);
+    gameStatus = getGameStatus[(<string>params) as GameStatusType];
+    result = gameStatus;
+  }
+
+  if (type === 'study') {
+    const paramNumber = <number>params;
+    const IS_POSITIVE_NUMBER = paramNumber > 0;
+    const IN_STUDY = arr.includes(paramNumber);
+    const NOT_IN_STUDY = !IN_STUDY;
+
+    if (!NOT_IN_STUDY) return;
```

Comment:
`IN_STUDY`가 사용되는 곳이 `NOT_IN_STUDY`변수 때문이라면 아래와 같이 사용해도 괜찮지 않을지?! 하는 생각이 드네요!
```suggestion
    const IN_STUDY = arr.includes(paramNumber);
    if (IN_STUDY) return;
```

Reply:
오오 맞네요!! 삭제 완료 >< ! 리뷰 감사해요 조이~